src/inspector_agent.cc

```diff
@@ -776,6 +770,16 @@ bool Agent::Start(const std::string& path,
     start_io_thread_async.data = this;
     // Ignore failure, SIGUSR1 won't work, but that should not block node start.
     StartDebugSignalHandler();
+
+    parent_env_->AddCleanupHook([](void* data) {
+      Environment* env = static_cast<Environment*>(data);
+
+      // This is global, will never get freed
+      env->CloseHandle(&start_io_thread_async, [](uv_async_t*) {
+        start_io_thread_async.data = nullptr;
+        CHECK(start_io_thread_async_initialized.exchange(false));
+      });
+    }, parent_env_);
```

Comment:
What happens when a SIGUSR1 signal is received at just the wrong time? It looks like this needs more synchronization with `StartIoThreadMain()` since that calls `Agent::RequestIoThreadStart()` which in turn calls `uv_async_send()`.

Reply:
@bnoordhuis Updated, PTAL – this should not call `uv_async_send()` on a closing/closed/uninitialized `start_io_thread_async` anymore